prowler/providers/aws/services/ec2/ec2_service.py

```diff
@@ -243,52 +242,38 @@ def __determine_public_snapshots__(self, snapshot):
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
-    def __describe_public_network_interfaces__(self, regional_client):
+    def __describe_network_interfaces__(self, regional_client):
         try:
             # Get Network Interfaces with Public IPs
             describe_network_interfaces_paginator = regional_client.get_paginator(
                 "describe_network_interfaces"
             )
             for page in describe_network_interfaces_paginator.paginate():
                 for interface in page["NetworkInterfaces"]:
-                    if interface.get("Association"):
-                        self.network_interfaces.append(
-                            NetworkInterface(
-                                public_ip=interface["Association"]["PublicIp"],
-                                type=interface["InterfaceType"],
-                                private_ip=interface["PrivateIpAddress"],
-                                subnet_id=interface["SubnetId"],
-                                vpc_id=interface["VpcId"],
-                                region=regional_client.region,
-                                tags=interface.get("TagSet"),
-                            )
-                        )
-
-        except Exception as error:
-            logger.error(
-                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
-            )
+                    eni = NetworkInterface(
+                        id=interface["NetworkInterfaceId"],
+                        association=interface.get("Association", {}),
+                        attachment=interface.get("Attachment", {}),
+                        private_ip=interface["PrivateIpAddress"],
+                        type=interface["InterfaceType"],
+                        subnet_id=interface["SubnetId"],
+                        vpc_id=interface["VpcId"],
+                        region=regional_client.region,
+                        tags=interface.get("TagSet"),
+                    )
+                    self.network_interfaces.append(eni)
+                    # Add Network Interface to Security Group
+                    # 'Groups': [
+                    #     {
+                    #         'GroupId': 'sg-xxxxx',
+                    #         'GroupName': 'default',
+                    #     },
+                    # ],
+                    for sg in interface.get("Groups", []):
+                        for security_group in self.security_groups:
+                            if security_group.id == sg["GroupId"]:
+                                security_group.network_interfaces.append(eni)
```

Comment:
Could you move this to a different function? One to get all the network interfaces ✅ other to link them with their security groups.

Reply:
Done!